### DISCUSSION TITLE
Cassandra write query in Apache Atlas which occurs via JanusGraph is failing

### DISCUSSION THREAD

**Discussion by mehtaanshul** [JanusGraph/janusgraph#2903] [Q&A]
Getting this exception when hitting an Atlas POST API which internally writes data to Cassandra via JanusGraph->
- JanusGraphException
   Could not commit transaction due to exception during persistence
   WriteFailureException
   Cassandra failure during write query at consistency QUORUM (2 responses were required but only 0 replica responded, 2 failed)

When checking Cassandra logs I found the following exception there - 
- java.lang.IllegalArgumentException: Mutation of 33.7662 MiB is too large for the maximum size of 16.000 MiB

and I found we can fix this at Cassandra level by following this - 
https://support.datastax.com/s/article/Mutation-of-x-bytes-is-too-large-for-the-maxiumum-size-of-y
by increasing the limit size using this property - `commitlog_segment_size_in_mb`

However, we do not want to increase this limit beyond a certain level and hence wanted to understand we can somehow fix this at the JanusGraph level. Please help!

Cassandra version - 2.1
JanusGraph version - 0.6.0

**Comment by farodin91** [comment]
With your provided the data, it is possible to test or fix the issue. It would be helpful to get full gremlin command for the transaction which produces this error. 

Just to mention, janusgraph 0.6 doesn't support officially Cassandra 2. It would to be helpful to reproduce this with Cassandra 3.
